### ISSUE TITLE
Make Entertainers an Object Type

### ISSUE THREAD

**Issue by That-Human-Being** [OpenRCT2/OpenRCT2#15326]
### Describe the feature

I know this would require the New Save Format, but selecting which entertainer types I want and (making custom entertainer types something people can make themselves) would be cool.

### Feature type

Other

**Comment by ocalhoun6** [comment]
While you're at it, it might be cool to be able to do that with other staff as well.

Imagine a 'plainclothes security guard' who uses standard guest sprites, for example.

And I'd be excited to see if anyone in the community starts making custom mechanic or handyman sprites. Would be really neat to see what they come up with.

Oh, and I suppose that some makers of custom parks might enjoy being able to have invisible mechanics/handymen that simply have no sprites, but still invisibly go around the park doing their jobs.

I'm not sure how feasible that is, though. Entertainers already come in multiple styles, so maybe it would be easier to have new entertainer objects than to have new handyman/mechanic/security guard objects.

**Comment by Broxzier** [comment]
> Though I suspect they'd turn into normal-looking guests whenever they board a ride, since the peeps riding a ride are actually part of the ride's sprites, right?

Correct. For staff this will be no issue. Some have unique sprites of course, like sweeping the paths for handymen and fixing a ride for mechanics, but those do not rely on objects.

**Comment by Triptych366** [comment]
If that's the case, perhaps we could have the entertainers do different things.  I could imagine a security guard arresting an guest for vandalizing a light pole, or something like that.  Maybe the entertainers could even entertain guest and guests start to gather creating a group.

**Comment by ocalhoun6** [comment]
>  I could imagine a security guard arresting a* guest for vandalizing a light pole

This would never happen, though, because peeps never vandalize anything if they're close to a security guard. So a security guard would never catch anybody.

-----

And, overall, these changes to actual staff behavior (and staff/guest interaction) should probably be spun off into its own issue. That's way outside the scope of what this issue is talking about.

**Comment by Triptych366** [comment]
> 
> 
> > I could imagine a security guard arresting a* guest for vandalizing a light pole
> 
> This would never happen, though, because peeps never vandalize anything if they're close to a security guard. So a security guard would never catch anybody.

Yeah, it would have to be like a guest gets labeled as the vandal, and if the security guard happens to run into him he'll get arrested.  Idk, but yeah I see you're point.

**Comment by Crazycolbster** [comment]
Did this change the way plugins work with hiring Entertainers? I just noticed today after a couple reports that the function that hires entertainers is no longer working in my plugin.

`var furry_type = Math.floor(Math.random() * 3);
            context.executeAction("staffhire",{autoPosition: true, staffType: 3, entertainerType: furry_type, staffOrders: 0} as StaffHireArgs);`

results in 
`Error: Invalid action parameters.`

**Comment by AaronVanGeffen** [comment]
Yes, the `StaffHire` game action was changed to require an `ObjectEntryIndex` instead of a `PeepAnimationType`. ObjectEntryIndex is not exposed to plugins, however, and clearly the API hadn't been updated to reflect this change, unfortunately. Not sure how we should approach this. Ideas/PRs welcome.

In the mean time, you can work around this by hiring a random entertainer, then working with their `.costume` and `.availableCostumes` properties instead.
